### PR TITLE
Densitysplit

### PR DESCRIPTION
**Pull request by sbird**
This moves one of the DhsmlDensityFactor variables into a local variable of density.c, further reducing the overhead of DENSITY_INDEPENDENT_SPH. No change to the hydro example, modest performance increase.

### CODE REVIEW

File: libgadget/density.c [hunk 1]
```diff
@@ -88,6 +88,11 @@ typedef struct {
 struct DensityPriv {
     MyFloat *Left, *Right;
     MyFloat (*Rot)[3];
+    /* This is the DhsmlDensityFactor for the pure density,
+     * not the entropy weighted density. It is used here and
+     * copied to DhsmlEgyDensityFactor if DensityIndependentSphOn = 1
+     * If DensityIndependentSphOn = 0 it is used to set DhsmlEgyDensityFactor. */
+    MyFloat * DhsmlDensityFactor;
```
Comment (rainwoodman) [inline libgadget/density.c]: This comment is a bit confusing. If DensityIndependentSphOn is false there is no DhsmlEgyDensityFactor in the force equations. Probably you mean DhsmlEgyDensityFactor and DhslmDensityFactor are the same if not doing densityindependentSph?

File: libgadget/density.c [hunk 2]
```diff
@@ -528,7 +551,13 @@ void density_check_neighbours (int i, TreeWalk * tw) {
             {
                 if(P[i].Type == 0 && fabs(P[i].NumNgb - desnumngb) < 0.5 * desnumngb)
                 {
-                    double fac = 1 - (P[i].NumNgb - desnumngb) / (NUMDIMS * P[i].NumNgb) * SPHP(i).DhsmlDensityFactor;
+                    MyFloat DensFac;
+                    if(DENSITY_GET_PRIV(tw)->DhsmlDensityFactor)
```
Comment (rainwoodman) [inline libgadget/density.c]: Maybe you want to use the master flag (cleary shows the motivation of the branching ) instead of a derived condition.

Comment (sbird) [reply libgadget/density.c]: Yes, I do!

File: libgadget/density.c (hunk 1 above)
Comment (sbird) [reply libgadget/density.c]: Thanks, clarified.

**Comment by rainwoodman** [comment]
Green. Merge?

**Comment by sbird** [comment]
Just checking the Lya example is still good...

**Comment by sbird** [comment]
Uh....I just found something worrying.

**Comment by sbird** [comment]
This passes the lya example too, merging.

**Pull request merged**